### ISSUE TITLE
stage2 parser crash on `extern\\`

### ISSUE THREAD

**Issue by xackus** [ziglang/zig#4163]
`extern\\` crashes the parser.
```
attempt to unwrap null
/home/mwa/code/zig/lib/std/zig/parse.zig:288:74: 0x23cbfa in std.zig.parse.parseTopLevelDecl (run)
            putBackToken(it, string_literal_node.cast(Node.StringLiteral).?.token);
                                                                         ^
/home/mwa/code/zig/lib/std/zig/parse.zig:113:34: 0x239377 in std.zig.parse.parseContainerMembers (run)
        if (try parseTopLevelDecl(arena, it, tree)) |node| {
                                 ^
/home/mwa/code/zig/lib/std/zig/parse.zig:70:43: 0x2367ae in std.zig.parse.parseRoot (run)
        .decls = try parseContainerMembers(arena, it, tree),
                                          ^
/home/mwa/code/zig/lib/std/zig/parse.zig:51:31: 0x209b49 in std.zig.parse.parse (run)
    tree.root_node = parseRoot(arena, &it, tree) catch |err| blk: {
                              ^
```

**Comment by Vexu** [comment]
Fixed by #3957

**Comment by mikdusan** [comment]
@Vexu,

On master d3e67d99216d3dd6c18259c17652fcad54aebc21:

#### source: `bug.zig`
```zig
extern\\
```

#### command: `cat bug.zig | zig fmt --stdin`
```
attempt to unwrap null
Unable to dump stack trace: FileNotFound
zsh: done       cat bug.zig | 
zsh: abort      zig fmt --stdin
```
